src/main/java/software/amazon/awssdk/http/apache/internal/conn/SdkTlsSocketFactory.java

```diff
@@ -50,9 +50,16 @@ public SdkTlsSocketFactory(final SSLContext sslContext, final HostnameVerifier h
     @Override
     protected final void prepareSocket(final SSLSocket socket) {
         // BEGIN GG MODIFICATIONS
-        SSLParameters params = new SSLParameters();
-        params.setApplicationProtocols(new String[]{"x-amzn-http-ca", "http/1.1"});
-        socket.setSSLParameters(params);
+        try {
+            SSLParameters params = new SSLParameters();
+            params.setApplicationProtocols(new String[]{"x-amzn-http-ca", "http/1.1"});
+            socket.setSSLParameters(params);
+        } catch (NoSuchMethodError e) {
+            log.debug(() -> "Unable to configure socket for ALPN. Ports other than 443 may still work.");
```

Comment:
Do we know here if they are using port 443 - and if so log at a higher level like warn?

Reply:
No, we can't tell because this is so low level. We're just making a socket which is later used to connect to some port. We don't have enough info to do anything smarter.